planner/core/find_best_task.go

```diff
@@ -888,20 +882,11 @@ func (ds *DataSource) convertToPartialIndexScan(prop *property.PhysicalProperty,
 }
 
 func (ds *DataSource) convertToPartialTableScan(prop *property.PhysicalProperty, path *util.AccessPath) (
-	tablePlan PhysicalPlan, partialCost float64, needExtraProj bool) {
+	tablePlan PhysicalPlan, partialCost float64) {
 	ts, partialCost, rowCount := ds.getOriginalPhysicalTableScan(prop, path, false)
-	if ds.tableInfo.IsCommonHandle {
-		commonHandle := ds.handleCols.(*CommonHandleCols)
-		for _, col := range commonHandle.columns {
-			if ts.schema.ColumnIndex(col) == -1 {
-				ts.Schema().Append(col)
-				ts.Columns = append(ts.Columns, col.ToInfo())
-				needExtraProj = true
-			}
-		}
-	}
 	rowSize := ds.TblColHists.GetAvgRowSize(ds.ctx, ds.TblCols, false, false)
 	sessVars := ds.ctx.GetSessionVars()
+	overwritePartialTableScanSchema(ds, ts)
```

Comment:
It's better to update the `rowSize` as well since we only return the handle column from the coprocessor to tidb now, for a more accurate cost estimation. You can do it in another PR if you prefer of course.